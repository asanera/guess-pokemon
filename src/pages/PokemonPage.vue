<template>
    <h1 v-if="!pokemonPropouse">Espere por favor...</h1>
    <div v-else>
        <h1>Â¿Quien es este pokemon?</h1>
        <PokemonPicture :pokemon-id="pokemonPropouse.id" :show-pokemon="showPokemon"></PokemonPicture>
        <PokemonOption :pokemons="pokemon" @selection-pokemon="checkAnswer"></PokemonOption>
        <template v-if="showAnswer">
            <h2 class="fade-in">{{message}}</h2>
            <button @click="newGame">Nuevo Juevo</button>
        </template>
        
    </div>

</template>

<script>
import PokemonPicture from '@/components/PokemonPicture.vue';
import PokemonOption from '@/components/PokemonOption.vue';
import getPokemonOptions from '@/helper/getPokemonOptions.js'

export default {
    name: "PokemonPage",
    components: { PokemonPicture, PokemonOption },
    data() {
        return {
            pokemon: [],
            pokemonPropouse: null,
            showPokemon: false,
            showAnswer: false,
            message: ""
        }
    },
    methods: {
        async mixPokemonArray(){
            this.pokemon = await getPokemonOptions();
            const rdnInt = Math.floor(Math.random() * this.MAX_POKEMONS_OPTIONS);
            this.pokemonPropouse = this.pokemon[rdnInt];
            
        },

        checkAnswer(selectedPokemonId){
            this.showPokemon = true;
            this.showAnswer = true;
            if(selectedPokemonId === this.pokemonPropouse.id) {
                this.message = `Correcto, ${this.pokemonPropouse.name}`
            }else {
                this.message = `Ops, era ${this.pokemonPropouse.name}`
            }
        },

        newGame() {
            this.mixPokemonArray()
            this.pokemon = []
            this.pokemonPropouse = null
            this.showPokemon = false
            this.showAnswer = false
            this.message = ""
        }
    },
    computed: {
        MAX_POKEMONS_OPTIONS: {
            get() {
                return 4
            }
        }
    },
    mounted(){
        this.mixPokemonArray();
    }
}
</script>

<style>

</style>