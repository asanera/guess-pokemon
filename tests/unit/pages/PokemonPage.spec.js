import PokemonPage from "@/pages/PokemonPage"
import { mount } from "@vue/test-utils"
import { pokemonsMock } from "../mocks/pokemons.mock"

describe('PokemonPage Component', () => {

    test('Debe de hacer match con el snapshot', () =>{
        const wrapper = mount(PokemonPage, {
            data() {
                return {
                    pokemon: pokemonsMock,
                    pokemonPropouse: pokemonsMock[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ""                    
                }
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Debe llamar mixPokemonArray al montar', () => {
        const mixPokemonArraySyp = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
        mount(PokemonPage);
        expect(mixPokemonArraySyp).toHaveBeenCalled()
    })
})