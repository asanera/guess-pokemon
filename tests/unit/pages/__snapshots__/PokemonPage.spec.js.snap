// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>Â¿Quien es este pokemon?</h1>
  <div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/5.svg" class="hidden-pokemon" alt="pokemon">
    <!--v-if-->
  </div>
  <div class="options-container">
    <ul>
      <li>pikachu</li>
      <li>ivysaur</li>
      <li>venusaur</li>
      <li>charmander</li>
    </ul>
  </div>
  <!--v-if-->
</div>
`;
